### PR TITLE
Monadic

### PR DESCRIPTION
**Pull request by Lokeh**
This changes `map` to be what is usually expected of a monadic interface, and introduces `flatMap` as a new name for the previous `map` function.

This is a breaking change. We should probably handle the versioning accordingly.

### CODE REVIEW

**Comment by wokalski** [comment]
Cool. Just two little things. 

1. Maybe we could use `>|=` to match [lwt](https://ocsigen.github.io/lwt/manual-draft/Lwt.html#Infix.VAL(>|=))
2. Maybe we should rename `flatMap` to `bind` to match lwt?

I'm also a bit concerned about `=<<`/`|<<`. It feels to me like unnecessary ambiguity especially for beginners who read the code. It's just pure sugar in the purest form and I'm not a fan of it.

**Comment by wokalski** [comment]
yo @Lokeh should I take over this?

**Comment by Lokeh** [comment]
Hey @wokalski, sorry for the delay. I've been very busy with work and some personal matters. I hope to get around to wrapping this up in the next week, but can't promise that. Feel free to pick it up if you'd like to

**Comment by Lokeh** [comment]
I made the changes you suggested. I personally feel that `flatMap` makes more sense than `bind`, but obviously plenty of people disagree with me 😆 .

If we want to capture similarities with lwt, then it makes sense to go with `bind`; if we want to capture people with experience in JS, I think it makes more sense to go with `flatMap`.

**Comment by wokalski** [comment]
I liked `andThen` and many people prefer flatMap from what I’ve seen. Pick one of the two. I’ll just blindly merge :).

File: src/vow.re [hunk 1]
```diff
@@ -28,8 +30,10 @@ module type ResultType = {
   type t 'value 'error 'status = vow (result 'value 'error) 'status;
   let return: 'value => t 'value 'error handled;
   let fail: 'error => t 'value 'error handled;
-  let map: ('a => t 'b 'error 'status) => t 'a 'error handled => t 'b 'error 'status;
-  let mapUnhandled: ('a => t 'b 'error 'status) => t 'a 'error unhandled => t 'b 'error unhandled;
+  let bind: ('a => t 'b 'error 'status) => t 'a 'error handled => t 'b 'error 'status;
+  let bindUnhandled: ('a => t 'b 'error 'status) => t 'a 'error unhandled => t 'b 'error unhandled;
+  let map: ('a => 'b) => t 'a 'error handled => t 'b 'error 'status;
```
Comment (wokalski) [inline src/vow.re]: This returns `t 'b handled`.

File: src/vow.re [hunk 2]
```diff
@@ -41,7 +45,7 @@ module type ResultType = {
     vow 'a 'status;
   module Infix: {
     let (>>=): t 'a 'error handled => ('a => t 'b 'error 'status) => t 'b 'error 'status';
-    let (=<<): ('a => t 'b 'error 'status) => t 'a 'error handled => t 'b 'error 'status;
+    let (>|=): t 'a 'error handled => ('a => 'b) => t 'b 'error 'status;
```
Comment (wokalski) [inline src/vow.re]: This also returns `t 'b 'error handled`

**Comment by Lokeh** [comment]
I did the needful! I chose flatMap - willing to be overruled / revisited at a later date :P

**Comment by wokalski** [comment]
Thank you! I'm so happy to have this repo in a "done" state.

**Pull request merged**